### PR TITLE
Fix Quick Open window to use EDSCALE and save the size between popups

### LINKED ISSUE #68363: Editor Command Palette Window Size is Small
### Godot version

All 4.0.beta

### System information

macOS Monterey 12.3.1

### Issue description

When opening the window `Quick Open…` or `Quick Open Scene…`, the size of the window is really small and doesn't allow to read the whole paths sometimes.
If I resize it, close it and reopen it, the default size is used again.

CleanShot 2022-11-07 at 11 57 30@2x

## How to enhance

Increase the window size by default or remember the size the user defined when resizing the window.

### Steps to reproduce

Open the `Quick Open` window by pressing `CMD+P`.

### Minimal reproduction project

Any projects can reproduce this "issue".

### PR DESCRIPTION
**Pull request by Chaosus**
Fix https://github.com/godotengine/godot/issues/68363

### CODE REVIEW

**Comment by Cronos87** [comment]
Does this fix the `Command Palette` too? Because it has the same issue as the `Quick Open` one, they have the same size.

**Comment by Chaosus** [comment]
> Does this fix the Command Palette too? Because it has the same issue as the Quick Open one, they have the same size.

Now, yes

**Comment by Cronos87** [comment]
Many thanks! That was very fast!

**Comment by MewPurPur** [comment]
Does this supercede #67517?

File: editor/editor_quick_open.cpp [hunk 1]
```diff
@@ -32,12 +32,18 @@
...
 
 void EditorQuickOpen::popup_dialog(const StringName &p_base, bool p_enable_multi, bool p_dontclear) {
 	base_type = p_base;
 	allow_multi_select = p_enable_multi;
 	search_options->set_select_mode(allow_multi_select ? Tree::SELECT_MULTI : Tree::SELECT_SINGLE);
-	popup_centered_clamped(Size2i(600, 440), 0.8f);
+	if (!sizes.has(p_base)) {
+		sizes.insert(p_base, Size2i(600, 440));
+	}
+	popup_centered_clamped(sizes[p_base] * EDSCALE, 0.8f);
```
Comment (KoBeWi) [inline editor/editor_quick_open.cpp:46]: The size is integer vector, so multiplication does not work for non-integer scales (e.g. 1.5).

Comment (Chaosus) [reply editor/editor_quick_open.cpp:46]: There are lots of places in the engine where this is already used, so I guess it doesn't matter:

image

Comment (KoBeWi) [reply editor/editor_quick_open.cpp:46]: Yes, the `sizes` map added in this PR uses Size2i. All examples you showed use Size2, so they are correct.

Comment (Chaosus) [reply editor/editor_quick_open.cpp:46]: Ah, sorry, I didn't notice...

Comment (Chaosus) [reply editor/editor_quick_open.cpp:46]: @KoBeWi Changed type to Size2

**Comment by KoBeWi** [comment]
For some reason the dialog increases with each open. Using scale 1.5.


https://user-images.githubusercontent.com/2223172/203370227-6db19749-0deb-44b6-89a1-3f2e2de81661.mp4

**Comment by Chaosus** [comment]
@KoBeWi Fixed, check again

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Cronos87** [comment]
Thank you, everyone! You all are awesome!

**Comment by Sauermann** [comment]
It looks like this PR introduced #69417.
As far as I can tell the issue is that `was_showed` is static, but the position is not saved during multiple calls to `EditorQuickOpen::popup_dialog`, because new instances of `EditorQuickOpen` are created.

**Pull request merged**